profiles/common/modules/features/nexteuropa_communities/nexteuropa_communities.module

```diff
@@ -1,7 +1,467 @@
+  $path = nexteuropa_multilingual_language_negotiation_split_suffix(request_path(), [$language]);
+  $path_args = explode('/', $path[1]);
+
+  // Look for the community keyword in the path.
+  if ($path_args[0] === $keyword) {
+    $community_path = drupal_lookup_path('source', $path_args[0] . '/' . $path_args[1], $language->language);
+    // If no path was found, then it's not a community in particular.
+    if ($community_path !== FALSE) {
+      $community_path = explode('/', $community_path);
+
+      // Check that the node acts as a group.
+      $node = node_load($community_path[1]);
+      if (og_is_group('node', $node)) {
+        return [
+          'node' => [
+            $node->nid,
+          ],
+        ];
+      }
+    }
+  }
+  return [];
+}
+
+/**
+ * Implements hook_token_info().
+ */
+function nexteuropa_communities_token_info() {
+  $info = [];
+
+  $info['tokens']['site']['nexteuropa_communities_keyword'] = [
+    'name' => t('NextEuropa Communities keyword'),
+    'description' => t('A token showing the url keyword triggering the community context detection.'),
+  ];
+
+  return $info;
+}
+
+/**
+ * Implements hook_tokens().
+ */
+function nexteuropa_communities_tokens($type, $tokens, $data, $options) {
+  $replacements = [];
+
+  switch ($type) {
+    case 'site':
+      foreach ($tokens as $name => $token) {
+        switch ($name) {
+          case 'nexteuropa_communities_keyword':
+            $replacements[$token] = variable_get('nexteuropa_communities_url_keyword', 'community');;
+            break;
+        }
+      }
+      break;
+  }
+
+  return $replacements;
+}
+
+/**
+ * Implements hook_permission().
+ */
+function nexteuropa_communities_permission() {
+  return array(
+    'access private area' => array(
+      'title' => t('Access private area'),
+      'description' => t('Allow to access to the private area.'),
+    ),
+  );
+}
+
+/**
+ * Implements hook_node_access().
+ */
+function nexteuropa_communities_node_access($node, $op, $account) {
+  if ($op != 'view') {
+    return;
+  }
+
+  if (variable_get('nexteuropa_communities_private_area', FALSE)) {
+    // The private area option is enabled.
+    // DENY access only on public contents,
+    // when the user has not the "access private area" permission.
+    $wrapper = entity_metadata_wrapper('node', $node);
+
+    // The node is a group.
+    if (!empty($wrapper->{OG_ACCESS_FIELD}) && $wrapper->{OG_ACCESS_FIELD}->value() == 0 && og_is_group('node', $node) && !user_access("access private area", $account)) {
+      return NODE_ACCESS_DENY;
+    }
+
+    // The node is a group content.
+    if (!empty($wrapper->{OG_CONTENT_ACCESS_FIELD}) && !user_access("access private area", $account)) {
+      $content_access = !empty($wrapper->{OG_CONTENT_ACCESS_FIELD}) ? $wrapper->{OG_CONTENT_ACCESS_FIELD}->value() : OG_CONTENT_ACCESS_DEFAULT;
+      switch ($content_access) {
+
+        case OG_CONTENT_ACCESS_PUBLIC:
+          return NODE_ACCESS_DENY;
```

Comment:
See 3 lines above:
```
`    if (!empty($wrapper->{OG_CONTENT_ACCESS_FIELD}) && !user_access("access private area", $account)) {
```
So, if the user don't have the permission "access private area", we refuse access for OG_CONTENT_ACCESS_PUBLIC. This is the goal the the "private area" functionality.